### ISSUE TITLE
How to extend es5 version and async

### ISSUE THREAD

**Issue by trevor403** [typicode/lowdb#81]
I made a clone of your repository and I changed the disk.js file completely to use the cordova file plugin. Would you say that it is okay to restructure the library to accept serialization overrides and read write overrides just like the es6 branch? Or would you suggest we just stick close to the es6 branch?

That being said, the es6 branch still relies on a sync read method. I use mine in browser obviously so I needed async read. I changed the lowdb constructor to bind the read promise to the db.object and return the promise. For the actual library you could return the actual db and have the Promise in a property? 

So basically we need full async support and to decide if we care about es5 anymore.

**Comment by typicode** [comment]
Hi @trevor403,

I would suggest sticking to `v0.11` (it's es6 but it's actually transpiled to es5) and create a third-party storage module for cordova file plugin that could be plugged in lowdb.

I've just added Promise support for `storage.read`:
https://github.com/typicode/lowdb/releases/tag/v0.11.2

For example, you could just write :

```js
// lowdb-cordova-file-storage.js
module.exports = {
  read: (source, deserialize) => // return a Promise
  write: (dest, obj, serialize) => // return a Promise
}
```

And use it with lowdb:

```js
low(source, {
  storage: require('lowdb-cordova-file-storage')
}).then(db => /* db is ready to be used */)
```

**Comment by trevor403** [comment]
Lovely solution! I'll make a mockup and close the issue.

**Comment by trevor403** [comment]
I was able to compile using babel and made a cordovaStorage object using cordova-file

**Comment by typicode** [comment]
Awesome :)
If you release it as a module on npm let me know.
